### ISSUE TITLE
Turn off or edit validation rules that mistakenly flag valid syntax

### ISSUE THREAD

**Issue by bbringhu** [AdobeDocs/adobe-markdown-authoring#26]
**MD049** - emphasis style - We don't enforce this rule.

![Screenshot 2023-03-22 at 11 13 29 AM](https://user-images.githubusercontent.com/6193209/226999634-f3028d54-21a0-4ed5-910a-a10ab85f641d.png)

**MD051** - link fragments should be valid - In this case, the deep link is valid. Let's just turn this off so that we don't have false positives.

![Screenshot 2023-03-22 at 11 17 36 AM](https://user-images.githubusercontent.com/6193209/227000049-8405ecc6-ad40-4d83-a981-516ef622ddcb.png)

**AM019** - Malformed link - This one isn't being applied properly. In this case, there is a set of parentheses after a link, which should be OK, but it's flagged.

![Screenshot 2023-03-22 at 11 37 45 AM](https://user-images.githubusercontent.com/6193209/227005069-946db2d7-32aa-4dc4-8aea-215f81e7e80f.png)

In this case, the image link is flagged as an error because there's hover text, which is supported.

![Screenshot 2023-03-22 at 2 43 20 PM](https://user-images.githubusercontent.com/6193209/227045202-f8ea5669-efd3-44c6-b63a-6534da819c88.png)

In this case, two links in the same sentence trigger an unwanted error:

![Screenshot 2023-03-22 at 2 49 42 PM](https://user-images.githubusercontent.com/6193209/227046679-942fde3f-a6b2-4098-927a-bc15fb4491d3.png)

**MD007** (_George_ - not fixed) - List indentation - We're being too strict in enforcing 4 spaces for lists. Most writers indent 3 spaces for nested content in numbered lists and only 2 spaces for nested comments in bullet lists. The example below is the correct syntax for a nested bullet list, but it's being flagged.

![Screenshot 2023-03-22 at 11 50 26 AM](https://user-images.githubusercontent.com/6193209/227007926-897f4cb6-6111-49d3-b2b6-e9f38d72a9b0.png)

**MD030** (_George_ - new, related to MD007)

We're mistakenly flagging this bullet list with only one space as an error. This bullet list would render fine. Don't enforce this rule.

![md030](https://user-images.githubusercontent.com/6193209/231522827-61e7595d-e80b-4ba7-8362-538f32b64848.png)

**AM014** - code block syntax - Flagged as error due to parameter French braces, which are supported.

![Screenshot 2023-03-22 at 2 45 37 PM](https://user-images.githubusercontent.com/6193209/227046004-2487bcc4-16a8-42d7-958e-afb80c9c44f5.png)

**AM004** - malformed table - In this example, the table format syntax is correct, but it's flagged as an error.

![Screenshot 2023-03-22 at 2 54 31 PM](https://user-images.githubusercontent.com/6193209/227047303-085c8420-6b8b-4d99-b5e5-5718c2e828a8.png)

**AM015** - HTML comment malformed - This comment is fine:

![Screenshot 2023-03-22 at 2 56 56 PM](https://user-images.githubusercontent.com/6193209/227047835-9a6fd578-d345-4ab6-ab30-d5719d6e399e.png)

CC: @geverit4

**Comment by geverit4** [comment]
I was able to address all of the AMXXX linter rules, but could not make code changes to the MDXXX rules.  I updated the configurations in the default config file shipped with the project where possible. 

## AM004 - malformed table

This was a conversion error to TypeScript.  

![Screenshot 2023-04-11 at 9 30 23 PM](https://user-images.githubusercontent.com/1187616/231324081-5f685e36-cc7e-4d4e-8a6c-565034f1078d.png)


## AM014 - code block syntax

Since the only thing this rule does is check for the presence of braces, it is now disabled in the main config. 
![Screenshot 2023-04-11 at 9 27 20 PM](https://user-images.githubusercontent.com/1187616/231323780-328a4736-69ed-4cb2-a1eb-978f5e42d07e.png)

## AM015 - HTML Comment malformed

This was a strange one.  An error was being added for no good reason.  

![Screenshot 2023-04-11 at 9 28 55 PM](https://user-images.githubusercontent.com/1187616/231323979-504f35b2-0b8d-4156-86d2-3dd41cb3907c.png)

## AM019 - Malformed Link

Deep links are now handled properly, as well as several other edge cases. 


![Screenshot 2023-04-11 at 9 31 39 PM](https://user-images.githubusercontent.com/1187616/231324276-b6f5e7c6-7a63-4f3f-94d3-87df66df7ee3.png)

**Comment by bbringhu** [comment]
@geverit4 - After testing, I found two things. MD007 still isn't working correctly, and MD030 is related. In both cases, we're flagging a single space after a bullet character as an error, but it's fine (and recommended). Everything else in this list looks great.

**Comment by bbringhu** [comment]
@geverit4 - Oops, one more. AM014 doesn't seem to be working properly in all cases. Example:

![Screenshot 2023-04-12 at 9 43 39 AM](https://user-images.githubusercontent.com/6193209/231525834-3554916f-bb6b-41fb-b3ac-2db6a7891766.png)

**Comment by geverit4** [comment]
@bbringhu 
## AM0014: 
The only thing that AM014 does is check for curly braces, so I disabled it in the package.json.  Oops - I actually didn't do that, just in my local settings.  That change will be in the next release. 

##MD007 and MD030:
This is a similar configuration issue. I misunderstood the problem.  I've now updated the markdownlint.config setting in the package.json. 

```JSON
  "markdownlint.config": {
    "line-length": false,
    ...
     "MD007": {
      "ul-indent": 3
    },
    ...
    "MD030": {
      "ul_multi": 1,
      "ol_multi": 1
    },
```



Can you attach an example?

**Comment by bbringhu** [comment]
@geverit4 - MD007 still isn't quite right. Did you add your local changes to 2.0.4?

![Screenshot 2023-04-17 at 10 53 27 AM](https://user-images.githubusercontent.com/6193209/232570219-20cca80a-c049-4328-bc1a-916927e11bb6.png)

**Comment by bbringhu** [comment]
I would rather turn off the rule than tell writers they're doing something wrong when they're doing exactly what we've told them to do.

```
* Bullet with with one space
* Another one

  * Sub-bullet list with one space
  * Another one
```

If we can't modify the rule to get the above to work, let's disable it.

**Comment by geverit4** [comment]
I set the indent property on the rule to 2 instead of 3, and this was the result:

![Screenshot 2023-04-17 at 4 21 55 PM](https://user-images.githubusercontent.com/1187616/232601466-410cfcd9-7e30-45ee-abff-103fc0a27a59.png)

Is that how it should work?  Here are the current settings for the extension: 
```JSON
"markdownlint.config": {
    "line-length": false,
    "AM001": false,
    "AM009": false,
    "AM011": false,
    "AM014": false,
    "AM015": false,
    "MD003": {
      "style": "atx"
    },
    "MD007": {
      "indent": 2
    },
    "MD009": false,
    "MD012": false,
    "MD014": false,
    "MD024": false,
    "MD025": {
      "front_matter_title": ""
    },
    "MD026": false,
    "MD027": false,
    "MD028": false,
    "MD030": {
      "ul_multi": 1,
      "ol_multi": 1
    },
    "MD033": {
      "allowed_elements": [
        "a",
        "b",
        "br",
        "caption",
        "code",
        "col",
        "colgroup",
        "div",
        "em",
        "I",
        "img",
        "li",
        "ol",
        "p",
        "pre",
        "s",
        "span",
        "strong",
        "sub",
        "sup",
        "table",
        "tbody",
        "td",
        "tfoot",
        "th",
        "thead",
        "tr",
        "u",
        "ul"
      ]
    },
    "MD036": false,
    "MD038": false,
    "MD039": false,
    "MD040": false,
    "MD045": false,
    "MD049": false,
    "MD051": false
  },
```

**Comment by bbringhu** [comment]
The problem is that we're being overly prescriptive. Writers can add 2 or 3 spaces, and both are fine. If we lock it in to 2 OR 3, we throw a false error for some people. What would be the drawback to turning off 007?

**Comment by geverit4** [comment]
The only drawback is that 007 would never be invoked, and even egregious violations would not be flagged.   

![Screenshot 2023-04-17 at 4 34 02 PM](https://user-images.githubusercontent.com/1187616/232604631-9676f86b-df85-43bd-82aa-e74a80fc6f25.png)

They could also click on the "Quick Fix" button to have VSCode set the indents to the indents setting throughout the document.

**Comment by bbringhu** [comment]
There's no way to allow either 2 or 3? We have to pick one?

**Comment by bbringhu** [comment]
If we have to choose, let's leave it as 3, since that's what it was before. I was really hoping that we could tell writers that if you see errors in VS Code, you must fix them to pass validation. Writers have no way to distinguish this error from a real error.

**Comment by bbringhu** [comment]
This looks great so far in testing. I'm going to keep this open for now in case I run into anything, but I'm pretty sure it's done. Fantastic!

**Comment by bbringhu** [comment]
I think MD007 is still enabled in 2.0.6. This rule is too fussy about spacing. We don't enforce MD007.

**Comment by bbringhu** [comment]
Fixed in 2.0.7
